otter/test/models/test_cass_models.py

```diff
@@ -3862,6 +3862,16 @@ def test_insert_empty(self):
             self.cache.insert_servers(self.dt, [], False),
             Effect(Constant(None)))
 
+    def test_insert_empty_delete(self):
+        """
+        `insert_servers` deletes servers when clear_others=True and does
+        nothing if passed list is empty
+        """
+        self.cache.delete_servers = lambda: Effect("delete")
+        eff = self.cache.insert_servers(self.dt, [], True)
+        self.assertEqual(eff.intent, "delete")
+        self.assertIsNone(resolve_effect(eff, None))
```

Comment:
This test ensures the _only_ `delete_servers` is called. It does that by first checking that effect to delete servers is returned and resolving that (i.e. running its callbacks) doesn't do anything. Does this help?